src/utils/hooks/usePaginateData.ts

```diff
@@ -11,35 +11,49 @@ interface UsePaginationDataArgs<T> {
 const DEFAULT_RESULTS_PER_PAGE = 10
 
 export const usePaginateData = <T>(args: UsePaginationDataArgs<T>) => {
-  const { data, keys, searchValue, resultsPerPage } = args
+  const {
+    data,
+    keys,
+    searchValue,
+    resultsPerPage = DEFAULT_RESULTS_PER_PAGE,
+  } = args
 
-  const fuseData = useRef(new Fuse(data, { keys }))
+  const fuseData = useRef(new Fuse(data, { keys, includeScore: true }))
 
   const [page, setPage] = useState(1)
+  const [totalPage, setTotalPage] = useState(
```

Comment:
Why is this state required?

Reply:
That was intentional because there was a weird state error with page, so adding that resolved it